src/lib.rs

```diff
@@ -247,14 +247,12 @@ impl<'w, 's, M: bevy_ecs::event::Event> IcedContext<'w, 's, M> {
 
         let cursor = {
             let window = self.windows.single();
-            match window.cursor_position() {
-                Some(position) => {
+            window.cursor_position().map_or_else(
+                || utils::process_touch_input(self).map_or(Cursor::Unavailable, Cursor::Available),
+                |position| {
                     Cursor::Available(utils::process_cursor_position(position, bounds, window))
-                }
-                None => utils::process_touch_input(self)
-                    .map(Cursor::Available)
-                    .unwrap_or(Cursor::Unavailable),
-            }
```

Comment:
Is there a performance advantage to `map_or_else`? As far as I can tell, [`map_or_else` compiles down to a `match` statement](https://doc.rust-lang.org/src/core/option.rs.html#1171-1174), and stylistically speaking, I prefer the `match` version.

Reply:
It does, but it is still "protected" inside the lazy evaluation of the closure. And that function does quite a lot of work. But I agree, match is cleaner, when you have a value that you already own or is already allocated

Reply:
So, why don't we leave it as the original `match` version?

Reply:
No problem, will do the change tomorrow

Reply:
af7ea2f @tasgon